src/api.ts

```diff
@@ -637,6 +657,10 @@ export function fernsRouter<T>(
 
       if (options.preUpdate) {
         try {
+          // TODO: Send flattened dot notation body to preUpdate, then merge the returned body
```

Comment:
So to make sure I'm tracking, all `preUpdate`s going forward then would be flattened, regardless of how much of the object was sent from the client? 

Looks like the commented code above mostly does this?

Reply:
Yup in the future, so right now you could get something like:

```{"foo.bar": true, bar: {baz: true}}``` 

Which is a little hard to work with if you want to make sure e.g. if foo.bar is updated, do XYZ. Ideal would be:

```{foo: {bar: true}, bar: {baz: true}}``` 

like you currently do. Flatten might be the wrong word here. I left the code above because it mostly works, but wound up breaking normal updates, so there's something off.